packages/block-library/src/query/index.php

```diff
@@ -123,3 +123,87 @@ function register_block_core_query() {
 	);
 }
 add_action( 'init', 'register_block_core_query' );
+
+/**
+ * Traverse the tree of blocks looking for any plugin block (i.e., a block from
+ * an installed plugin) inside a Query block with the enhanced pagination
+ * enabled. If at least one is found, the enhanced pagination is effectively
+ * disabled to prevent any potential incompatibilities.
+ *
+ * @since 6.4.0
+ *
+ * @param array $parsed_block The block being rendered.
+ *
+ * @return string Returns the parsed block, unmodified.
+ */
+function block_core_query_disable_enhanced_pagination( $parsed_block ) {
+	static $enhanced_query_stack   = array();
+	static $dirty_enhanced_queries = array();
+	static $render_query_callback  = null;
+
+	$block_name = $parsed_block['blockName'];
+
+	if (
+		'core/query' === $block_name &&
+		isset( $parsed_block['attrs']['enhancedPagination'] ) &&
+		true === $parsed_block['attrs']['enhancedPagination'] &&
+		isset( $parsed_block['attrs']['queryId'] )
+	) {
+		$enhanced_query_stack[] = $parsed_block['attrs']['queryId'];
+
+		if ( ! isset( $render_query_callback ) ) {
+			/**
+			 * Filter that disables the enhanced pagination feature during block
+			 * rendering when a plugin block has been found inside. It does so
+			 * by adding an attribute called `data-wp-navigation-disabled` which
+			 * is later handled by the front-end logic.
+			 *
+			 * @param string   $content  The block content.
+			 * @param array    $block    The full block, including name and attributes.
+			 *
+			 * @return string Returns the modified output of the query block.
```

Comment:
```suggestion
			 * @return string Returns the modified output of the query block.
```

As above.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/55714/commits/221b70b90d2eaddb216886fe8d888251679beb0a.